### PR TITLE
chore: add eslint config files

### PR DESCRIPTION
**Pull request by will-t-harris**
- Add initial config files for backend, frontend, and RN projects. These configs are collated from existing Yeti projects and are meant to be a good starting point for iterating on what our "official" eslint config should be.

### CODE REVIEW

**Review by rmutter — APPROVED** [review]

**Review by pfuscoe — APPROVED** [review]

File: config/eslint/backend/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "default-case": 2,
+    "interface-name": 0,
+    "object-literal-sort-keys": 0,
+    "ordered-imports": 0,
+    quotemark: 0,
+    "no-console": 0,
+    "no-undef": 0,
+    "no-var": 2,
+    "@typescript-eslint/indent": [2, 2],
+    "@typescript-eslint/no-explicit-any": 0,
```
Comment (will-t-harris) [inline config/eslint/backend/.eslintrc.js:28]: Change this to be on

**Review by dgillen27 — APPROVED** [review]

File: config/eslint/frontend/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "react/jsx-props-no-spreading": 0,
+    "react/destructuring-assignment": [2, "always"],
+    "react/no-deprecated": 2,
+    "react/no-unsafe": 2,
+    "react/no-unused-state": 2,
+    "react/jsx-pascal-case": 2,
+    "react/react-in-jsx-scope": 0,
+    "react-hooks/exhaustive-deps": 0,
+    "@emotion/syntax-preference": [2, "object"],
+    "jsx-a11y/no-static-element-interactions": 0,
```
Comment (dgillen27) [inline config/eslint/frontend/.eslintrc.js:29]: Should have mentioned earlier, I think this is something we should consider turning on for accessibility purposes

Comment (dgillen27) [reply config/eslint/frontend/.eslintrc.js:29]: another rule that is related that might be worth adding https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/click-events-have-key-events.md

**Review by dgillen27 — APPROVED** [review]

File: config/eslint/frontend/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "react/require-default-props": "off",
+    "no-nested-ternary": "off",
+    "prettier/prettier": 1,
```
Comment (pfuscoe) [inline config/eslint/frontend/.eslintrc.js:32]: I believe 30-32 came from the Fandom snippet
- `require-default-props` - prefer off, but open to discuss
- `no-nested-ternary` - prefer off, but open to discuss
- `prettier/prettier` - were we thinking this maybe should not be a linting rule, but instead a pre-commit hook? I'm definitely fine with it as a warning here. I'd be open to setting it to error, but sometimes that can feel tedious when testing locally

Comment (will-t-harris) [reply config/eslint/frontend/.eslintrc.js:32]: - agreed on `require-default-props`
- `no-nested-ternary` - I do tend to think that nested ternaries make logic more difficult to parse in general, but don't feel too strongly about this if others want to leave this disabled.
- `prettier/prettier` - I think we can leave this as a warning. From what I can tell in [the docs](https://github.com/prettier/eslint-plugin-prettier#recommended-configuration) `plugin:prettier/recommended` should handle separating prettier and eslint for us.

File: config/eslint/frontend/.eslintrc.js (hunk 2 above)
Comment (will-t-harris) [reply config/eslint/frontend/.eslintrc.js:29]: > another rule that is related that might be worth adding [https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/click-events-have-key-events.md](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/click-events-have-key-events.md?rgh-link-date=2022-05-17T15%3A58%3A10Z)

I think the move here is to include the `jsx-ally` plugin, I think it'll have that rule (and others) on by default?

File: config/eslint/frontend/.eslintrc.js (hunk 3 above)
Comment (Cscharf91) [reply config/eslint/frontend/.eslintrc.js:31]: I'll +1 to `no-nested-ternary` off, but I could also go either way on it. The way prettier formats them helps slightly with readability.

In my opinion cluttered if/else logic is almost as bad, but admittedly a bit less confusing than nested ternaries.

**Pull request merged**